### PR TITLE
Restructuring

### PR DESCRIPTION
**Pull request by emma-miler**
Restructuring of the files to help organisation of the project.
Moves source files out of the root directory of NorthstarDLL and sorts them into folders.
This is partly preparation for the new toolchain.

### CODE REVIEW

File: NorthstarDLL/core/structs.h [hunk 1]
```diff
@@ -46,14 +46,22 @@ OFFSET_STRUCT(Name)
...
 #define PROBE(x) x, 1
 
+#ifdef __clang__
+#define CHECK(...) CHECK_N(__VA_ARGS__, 0)
+#define DO_PROBE(offset) PROBE_PROXY(OFFSET_##offset) // concatenate prefix with offset
+#define PROBE_PROXY(...) PROBE_PRIMITIVE(__VA_ARGS__) // expand arguments
+#define PROBE_PRIMITIVE(x) PROBE_COMBINE_ x // merge
+#define PROBE_COMBINE_(...) PROBE(~) // if merge successful, expand to probe
+#else
 #define MSVC_VA_ARGS_WORKAROUND(define, args) define args
```
Comment (pg9182) [inline NorthstarDLL/core/structs.h:56]: Stray change?

**Review by pg9182 — APPROVED** [review]
LGTM, builds fine.

File: NorthstarDLL/core/structs.h (hunk 1 above)
Comment (emma-miler) [reply NorthstarDLL/core/structs.h:56]: Ah, leftover from the other PR. Will remove.

**Comment by GeckoEidechse** [comment]
Quoting @BobTheBob9 

> yeah just looked, looks good

**Pull request merged**